src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -34,26 +36,49 @@ class TemplateCacheWarmer implements CacheWarmerInterface, ServiceSubscriberInte
     public function __construct(
         private ContainerInterface $container,
         private iterable $iterator,
+        private ?CacheInterface $cache = null,
     ) {
     }
 
     public function warmUp(string $cacheDir, ?string $buildDir = null): array
     {
         $this->twig ??= $this->container->get('twig');
 
-        foreach ($this->iterator as $template) {
-            try {
-                $this->twig->load($template);
-            } catch (Error) {
+        $originalCache = $this->twig->getCache();
+        if ($originalCache instanceof NullCache) {
+            // There's no point to warm up a cache that won't be used afterward
+            return [];
+        }
+
+        if (null !== $this->cache) {
+            if (!$buildDir) {
                 /*
-                 * Problem during compilation, give up for this template (e.g. syntax errors).
-                 * Failing silently here allows to ignore templates that rely on functions that aren't available in
-                 * the current environment. For example, the WebProfilerBundle shouldn't be available in the prod
-                 * environment, but some templates that are never used in prod might rely on functions the bundle provides.
-                 * As we can't detect which templates are "really" important, we try to load all of them and ignore
-                 * errors. Error checks may be performed by calling the lint:twig command.
+                 * Skip the cache warming, because the ChainCache relies on the $buildDir to be writeable, which is
+                 * probably not the case.
```

Comment:
The `$buildDir` parameter is only set when the container is initially built; which is not always related to whether it's writable or not. The term "probably" is too vague.

I suggest this comment:

```suggestion
                 * The cache has already been warmup during the build of the container,
                 * when $buildDir was set.
```